src/AppRoot.tsx

```diff
@@ -793,11 +841,30 @@ class AppRoot extends React.Component<Props, State> {
       isVoterCardPrinted,
       isRecentVoterPrint,
       machineId,
+      hasCardReaderAttached,
+      hasChargerAttached,
+      hasLowBattery,
+      hasPrinterAttached,
+      hasPrinterError,
       precinctId,
       tally,
       userSettings,
       votes,
     } = this.state
+    if (hasLowBattery && !hasChargerAttached) {
```

Comment:
All these will stop a voter who is in the middle of voting, right? Is that the behavior we want?

Reply:
Yes. For loss of any functionality which will not allow a voter to complete their vote, we display one of these error screens. 

In the case of this specific error, if the charger is not plugged in and `!hasLowBattery`, the voter will see a non-blocking notification warning (TBD) on the Insert Card screen which is informational is design and has copy like: "This machine is on battery power. Please ask a poll worker to plug-in the power cord for this machine."